hopsworks-common/src/main/java/io/hops/hopsworks/common/dao/jupyter/config/JupyterFacade.java

```diff
@@ -120,13 +120,28 @@ public JupyterProject findByUser(String hdfsUser) {
     }
     return res2;
   }
-  
-  public void stopServer(String hdfsUser) {
-    if (Strings.isNullOrEmpty(hdfsUser)) {
-      throw new IllegalArgumentException("hdfsUser was not provided.");
+
+  public JupyterProject findByUser(String hdfsUser) {
+    HdfsUsers res = null;
+    TypedQuery<HdfsUsers> query = em.createNamedQuery(
+      "HdfsUsers.findByName", HdfsUsers.class);
+    query.setParameter("name", hdfsUser);
+    try {
+      res = query.getSingleResult();
+    } catch (NoResultException e) {
+      return null;
     }
-    JupyterProject jp = this.findByUser(hdfsUser);
-    remove(jp);
+    JupyterProject res2 = null;
+    TypedQuery<JupyterProject> query2 = em.createNamedQuery(
+      "JupyterProject.findByHdfsUserId", JupyterProject.class);
+    query2.setParameter("hdfsUserId", res.getId());
+    try {
+      res2 = query2.getSingleResult();
```

Comment:
@robzor92 better return query2 directly here and at line 144 return `null`. Also rename query2 to queryJupyterProject or something similar

Reply:
Fixed